### PR TITLE
Ncl 5628

### PR DESCRIPTION
**Pull request by DnsZhou**
### Checklist:

* [ ] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/pnc/wiki/Changelog) for your change if user-facing?
* [ ] Have you added unit tests for your change?

### CODE REVIEW

File: ui/app/common/pnc-client/resources/Environment.js [hunk 1]
```diff
@@ -31,7 +31,7 @@
     'restConfig',
     'ENVIRONMENT_PATH',
     function($resource, restConfig, ENVIRONMENT_PATH) {
-      var ENDPOINT = restConfig.getPncUrl() + ENVIRONMENT_PATH;
+      var ENDPOINT = restConfig.getPncRestUrl() + ENVIRONMENT_PATH;
```
Comment (matedo1) [inline ui/app/common/pnc-client/resources/Environment.js:34]: I guess `Environment.js` file is completely refactored right now? so it should be renamed to `EnvironmentResource.js` (including factory method name and its usages).

File: ui/app/common/pnc-client/resources/BuildConfiguration.js [hunk 2]
```diff
@@ -89,7 +89,7 @@
       });
 
       resource.getSupportedGenericParameters = function() {
-        return $http.get(restConfig.getPncUrl() + '/build-configurations/supported-generic-parameters').then(function (r) {
+        return $http.get(restConfig.getPncRestUrl() + '/build-configs/supported-parameters').then(function (r) {
```
Comment (matedo1) [inline ui/app/common/pnc-client/resources/BuildConfiguration.js:92]: `BuildConfiguration.js` is deprecated as `BuildConfigResource.js` is already available, the correct solution should implement missing methods there as start using `BuildConfigResource` factory method.

**Comment by DnsZhou** [comment]
Hi @matedo1, thanks for your review, I will fix that accordingly.

**Comment by DnsZhou** [comment]
Due to some conflicts of the code Base, I closed this PR and raised another PR to handle it: https://github.com/project-ncl/pnc/pull/2978

**Pull request closed without merging**